internal/controller/dragonfly_instance.go

```diff
@@ -168,15 +168,25 @@ func (dfi *DragonflyInstance) masterExists(ctx context.Context) (bool, error) {
 }
 
 func (dfi *DragonflyInstance) getMasterIp(ctx context.Context) (string, error) {
-	dfi.log.Info("retrieving ip of the master")
+	dfi.log.Info("retrieving IP of the master")
 	pods, err := dfi.getPods(ctx)
 	if err != nil {
 		return "", err
 	}
 
 	for _, pod := range pods.Items {
-		if pod.Status.Phase == corev1.PodRunning && pod.Status.ContainerStatuses[0].Ready && pod.Labels[resources.Role] == resources.Master {
-			return pod.Status.PodIP, nil
+		if pod.Status.Phase == corev1.PodRunning &&
+			pod.Status.ContainerStatuses[0].Ready &&
+			pod.Labels[resources.Role] == resources.Master {
+
+			masterIp, hasMasterIp := pod.Annotations[resources.MasterIp]
```

Comment:
If its a master, we don't expect the annotation to be set here right? I'm not sure how this specific change is useful 🤔

Reply:
Yes, i'm agree, I spot this too, I maybe broke it during the fix/my several refacto !